Models/PFUser+ExtendedUser.m

```diff
@@ -47,7 +47,9 @@ - (void)removeFavorite:(GMSPlace*)place {
            {
                if(result != nil)
                {
-                   [self.favorites removeObject:result];
+                   NSMutableArray *mutableFavorites = [NSMutableArray arrayWithArray:self.favorites];
+                   [mutableFavorites removeObject:result];
+                   self.favorites = mutableFavorites;
```

Comment:
I think you can actually convert the mutable array to an immutable one with [mutableFavorites copy]